textile/features.textile

```diff
@@ -1305,6 +1305,16 @@ h4. Message
 ** @(TM2e)@ @encoding@ string
 ** @(TM2i)@ @extras@ JSON-encodable object, used to contain any arbitrary key value pairs which may also contain other primitive JSON types, JSON-encodable objects or JSON-encodable arrays. The @extras@ field is provided to contain message metadata and/or ancillary payloads in support of specific functionality, e.g. push. Each of these supported extensions is documented separately; for 1.1 the only supported extension is @push@, via the @extras.push@ member; 1.2 adds the @delta@ extension whose keys and values are described by the attributes of the type @DeltaExtras@, and the @headers@ extension, which contains arbitrary @string->string@ key-value pairs, settable at publish time, and @ref@ whose keys and values are described by the attributes of the type @ReferenceExtras@. Unless otherwise specified, the client library should not attempt to do any filtering or validation of the @extras@ field itself, but should treat it opaquely, encoding it and passing it to realtime unaltered.
 ** @(TM2f)@ @timestamp@ time in milliseconds since epoch. If a message received from Ably does not contain a @timestamp@, it should be set to the @timestamp@ of the encapsulating @ProtocolMessage@
+** @(TM2j)@ @action@ enum
+** @(TM2K)@ @serial@ string - an opaque string that uniquely identifies the message.
+** @(TM2L)@ @refSerial@ string - an opaque string that uniquely identifies some referenced message.
+** @(TM2M)@ @refType@ string - an opaque string that identifies the type of this reference.
+** @(TM2N)@ @operation@ - object that may contain the following `optional` attributes;
+*** @(TM2N1)@ @clientId@ string
+*** @(TM2N2)@ @description@ string
+*** @(TM2N3)@ @metadata@ object - used to contain any arbitrary key value pairs of type string.
+** @(TM2O)@ @updatedAt@ time in milliseconds since epoch at which an operation occurred. This field is always populated on messages received with an @action@ of @MESSAGE_UPDATE@ or @MESSAGE_DELETE@..
+** @(TM2P)@ @updateSerial@ string - an opaque string that uniquely identifies the operation. This field is always populated on messages received with an @action@ of @MESSAGE_UPDATE@ or @MESSAGE_DELETE@.
```

Comment:
- the spec uses the convention that the letters in spec items other than the initial section name (eg the p in TM2p) is lowercase
- you reference a TM5 from the IDL that lists the actions, where is it?

Reply:
I've made the case changes and included TM5.